src/SignalR/clients/java/signalr/src/test/java/com/microsoft/signalr/HubConnectionTest.java

```diff
@@ -1576,11 +1577,12 @@ public void receiveHandshakeResponseAndMessage() {
             value.getAndUpdate((val) -> val + 1);
         });
 
+        SingleSubject<String> handshakeMessageTask = mockTransport.getNextSentMessage();
         // On start we're going to receive the handshake response and also an invocation in the same payload.
         hubConnection.start();
-        mockTransport.getStartTask().timeout(1, TimeUnit.SECONDS).blockingAwait();
+        String sentMessage = handshakeMessageTask.timeout(1, TimeUnit.SECONDS).blockingGet();
```

Comment:
Do these short 1 second timeouts ever cause flakiness? If not, I'm impressed.

Reply:
There were one or two flaky tests in java but I don't think they were ever related to these timeouts.

And actually, yesterday I was looking into them and there is a pretty good reason these never trigger. We can talk about it later today.